### PR TITLE
Add doc for openEuler 22.03

### PR DESCRIPTION
**Pull request by Yikun**
We fork the doc from existing vesion`cp -R rocky8 openEuler22.03` and apply https://github.com/openhpc/ohpc/pull/1748/commits/bb0eb6e28cc6f15b361433c61fbf70dafae86300 specific changes:

- Apply openEuler changes in steps.tex
- Add openEuler changes to docs.spec
- Skip doc build because the doc.spec depends on a manual doc tar.

https://github.com/openhpc/ohpc/blob/a12ce6765a90b5ad5a8a1709b6958cbe28b48701/components/admin/docs/SPECS/docs.spec#L20

I validate it by build doc locally:
- [aarch64_warewulf_openpbs_steps.pdf](https://github.com/openhpc/ohpc/files/11051844/aarch64_warewulf_openpbs_steps.pdf)
- [aarch64_warewulf_slurm_steps.pdf](https://github.com/openhpc/ohpc/files/11051845/aarch64_warewulf_slurm_steps.pdf)
- [x86_64_warewulf_openpbs_steps.pdf](https://github.com/openhpc/ohpc/files/11051850/x86_64_warewulf_openpbs_steps.pdf)
- [x86_64_warewulf_slurm_steps.pdf](https://github.com/openhpc/ohpc/files/11051855/x86_64_warewulf_slurm_steps.pdf)

### CODE REVIEW

**Comment by Yikun** [comment]
Fix suse CI error in https://github.com/openhpc/ohpc/pull/1749

**Comment by Yikun** [comment]
I will rebase soon

**Comment by adrianreber** [comment]
This feels wrong. You probably should not copy files from common. That should be a symbolic link.

**Comment by Yikun** [comment]
> This feels wrong. You probably should not copy files from common. That should be a symbolic link.

Yes, I also found this, I'm re-making this PR. : )

**Comment by Yikun** [comment]
Validate doc with https://github.com/openhpc/ohpc/pull/1750 :

- [aarch64_warewulf_openpbs_steps.pdf](https://github.com/openhpc/ohpc/files/11051844/aarch64_warewulf_openpbs_steps.pdf)
- [aarch64_warewulf_slurm_steps.pdf](https://github.com/openhpc/ohpc/files/11051845/aarch64_warewulf_slurm_steps.pdf)
- [x86_64_warewulf_openpbs_steps.pdf](https://github.com/openhpc/ohpc/files/11051850/x86_64_warewulf_openpbs_steps.pdf)
- [x86_64_warewulf_slurm_steps.pdf](https://github.com/openhpc/ohpc/files/11051855/x86_64_warewulf_slurm_steps.pdf)

Ready for review.

File: docs/recipes/install/common/openeuler_repos.tex
Comment (martin-g) [inline docs/recipes/install/common/openeuler_repos.tex:2]: There was a discussion recently on the mailing list that xCAT is no more supported. Since 2.3 IIRC.
But it could be cleaned up in a separate PR.

**Review by martin-g — APPROVED** [review]

File: docs/recipes/install/common/openeuler_repos.tex
Comment (Yikun) [reply docs/recipes/install/common/openeuler_repos.tex:2]: image

Yes, I think we can remove it in separate PR to cleanup `isxCAT` if needed, BTW rendered pdf also didn't include this becuase `isxCAT` is false in openEuler.

**Comment by Yikun** [comment]
@adrianreber Would you mind to take a look? I think it's ready to go.

**Comment by adrianreber** [comment]
> @adrianreber Would you mind to take a look? I think it's ready to go.

I will probably wait for documentation related changes a bit longer before merging them. From my point of view this is currently blocked by the test system. Once that is setup correctly for 3.x I would merge this PR.

**Comment by Yikun** [comment]
@adrianreber OK, thanks!

File: components/admin/docs/SPECS/docs.spec
Comment (martin-g) [inline components/admin/docs/SPECS/docs.spec:112]: I think we need an `%if 0%{?openEuler}` at line 70 to define `source_path` to be `docs/recipes/install/openeuler22.03`

Comment (Yikun) [reply components/admin/docs/SPECS/docs.spec:112]: Good catch, addressed

**Comment by martin-g** [comment]
@mghpcsim This is the PR for openEuler docs that generate the recipe.sh.

I also have a PR in my fork that is based on top of this PR and adds execution of the recipe in a self-hosted Github runner - https://github.com/martin-g/openeuler-ohpc/pull/2 . 
It is not very stable though because the compute nodes sometime need more time to become available after the power reset. Also I wasn't able to find a good way to execute the smoke test (the hello world MPI task) with `prun` without creating a new shell session. I wanted to check how it is being tested currently in Jenkins.

**Comment by adrianreber** [comment]
This works in CI and is ready to be merged. There is still a `.DS_Store` part of this PR. Please remove that file.

I think the last two commits can be squashed.

**Comment by Yikun** [comment]
@adrianreber done ^

- Remove `.DS_Store` and fix tail space
- Squash last two commits into one.

**Comment by adrianreber** [comment]
Thanks for the update.

Do you know why there are packages mentioned with `gnu9` and `arm1`. We do not have the ARM HPC compiler for openEuler and `gnu9` is also not available for 3.x.

Can that be removed or is that content automatically created?

**Comment by Yikun** [comment]
As I mentioned in first commit, the openEuler doc is [initialized](https://github.com/openhpc/ohpc/pull/1748/commits/a8f9b725a5da0677644651f8894eb008865f0d1a) from exsiting doc.

Yes, seems we need remove `gnu9`/`arm1-`/`intel-` in openEuler, and we also need to remove `gnu9` for other OS? But frankly, I don't know how to generate these conent automatically.

- According some history like https://github.com/openhpc/ohpc/commit/5fe2484d5badb51a0a1d5c8c8db560f1d42e5308 seems we change `gnu9` to `gnu12` manully.
- And seems there are also some script to help generate these doc: https://github.com/openhpc/ohpc/commits/3.x/docs/recipes/install/common/build_tables.pl

**Comment by adrianreber** [comment]
> As I mentioned in first commit, the openEuler doc is [initialized](https://github.com/openhpc/ohpc/pull/1748/commits/a8f9b725a5da0677644651f8894eb008865f0d1a) from exsiting doc.
> 
> Yes, seems we need remove `gnu9`/`arm1-`/`intel-` in openEuler, and we also need to remove `gnu9` for other OS? But frankly, I don't know how to generate these conent automatically.
> 
>     * According some history like [5fe2484](https://github.com/openhpc/ohpc/commit/5fe2484d5badb51a0a1d5c8c8db560f1d42e5308) seems we change `gnu9` to `gnu12` manully.
> 
>     * And seems there are also some script to help generate these doc: https://github.com/openhpc/ohpc/commits/3.x/docs/recipes/install/common/build_tables.pl

Okay. Let's merge this as is and maybe somebody who knows the setup better can help us at some point.

**Pull request merged**
